include/matrix_operations/EigenvaluesProblems.h

```diff
@@ -168,8 +168,9 @@ class EigenvaluesProblems<NT, Eigen::Matrix<NT,Eigen::Dynamic,Eigen::Dynamic>, E
         Spectra::DenseCholesky<NT> Bop(-A);
 
         // Construct generalized eigen solver object, requesting the largest three generalized eigenvalues
+        int ncv = std::min(std::max(10, matrixDim/20), matrixDim);
```

Comment:
Could you please comment on where those numbers come from? `10, 20, 15` ?

Reply:
This is an empirical value I gave to @atrayees. We left as a task after the R interface to tune this implementation (not only ncv value).

Reply:
Thanks @TolisChal for the explanations. @atrayees could you please add a comment with this explanation and mention that as a TODO?

Reply:
yes, done